src/Compilers/VisualBasic/Portable/Declarations/DeclarationKind.vb

```diff
@@ -1,17 +1,42 @@
 ﻿' Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
 
-Imports System.Collections.Generic
-Imports System.Collections.ObjectModel
-Imports System.Linq
-Imports System.Text
-Imports System.Threading
-Imports Microsoft.CodeAnalysis.Text
-Imports Microsoft.CodeAnalysis.VisualBasic
-Imports Microsoft.CodeAnalysis.VisualBasic.Symbols
-Imports Microsoft.CodeAnalysis.VisualBasic.Syntax
+Imports System.Runtime.CompilerServices
 
 Namespace Microsoft.CodeAnalysis.VisualBasic.Symbols
 
+    Friend Module EnumConversions
+        <Extension()>
+        Public Function ToTypeKind(kind As DeclarationKind) As TypeKind
+            Select Case kind
+                Case DeclarationKind.Class,
+                     DeclarationKind.Script,
+                     DeclarationKind.ImplicitClass
+                    Return TypeKind.Class
+
+                Case DeclarationKind.Submission
+                    Return TypeKind.Submission
+
+                Case DeclarationKind.Delegate
+                Case DeclarationKind.EventSyntheticDelegate
+                    Return TypeKind.Delegate
+
+                Case DeclarationKind.Enum
+                    Return TypeKind.Enum
+
+                Case DeclarationKind.Interface
+                    Return TypeKind.Interface
+
+                Case DeclarationKind.Structure
+                    Return TypeKind.Structure
+
+                Case DeclarationKind.Module
+                    Return TypeKind.Module
+            End Select
+
+            Throw ExceptionUtilities.UnexpectedValue(kind)
```

Comment:
this code does not feel VB specific (unless VB has it's only completley separate DeclarationKind enum.

Reply:
> unless VB has it's only completley separate DeclarationKind enum

This is indeed the case

Reply:
that's fine then.  thanks.